src/image_builder.py

```diff
@@ -155,109 +159,259 @@ def image_builder(buildspec):
             "labels": labels,
             "extra_build_args": extra_build_args
         }
-
-        image_object = DockerImage(
+        
+        # Create pre_push stage docker object
+        pre_push_stage_image_object = DockerImage(
             info=info,
             dockerfile=image_config["docker_file"],
             repository=image_repo_uri,
             tag=image_tag,
             to_build=image_config["build"],
+            stage=constants.PRE_PUSH_STAGE,
             context=context,
         )
 
-        IMAGES.append(image_object)
+        ##### Create Common stage docker object #####
+        # If for a pre_push stage image we create a common stage image, then we do not push the pre_push stage image
+        # to the repository. Instead, we just push its common stage image to the repository. Therefore,
+        # inside function get_common_stage_image_object we make pre_push_stage_image_object non pushable.
+        common_stage_image_object = generate_common_stage_image_object(pre_push_stage_image_object)
+
+        PRE_PUSH_STAGE_IMAGES.append(pre_push_stage_image_object)
+        COMMON_STAGE_IMAGES.append(common_stage_image_object)
+        FORMATTER.separator()
 
     FORMATTER.banner("DLC")
-    FORMATTER.title("Status")
 
-    THREADS = {}
+    # Standard images must be built before example images
+    # Example images will use standard images as base
+    # Common images must be built at the end as they will consume respective standard and example images
+    standard_images = [image for image in PRE_PUSH_STAGE_IMAGES if "example" not in image.name.lower()]
+    example_images = [image for image in PRE_PUSH_STAGE_IMAGES if "example" in image.name.lower()]
+    ALL_IMAGES = PRE_PUSH_STAGE_IMAGES + COMMON_STAGE_IMAGES
+    IMAGES_TO_PUSH = [image for image in ALL_IMAGES if image.to_push and image.to_build]
```

Comment:
Can the `IMAGES_TO_PUSH` be `COMMON_STAGE_IMAGES` ?

Reply:
Synced up